### PR TITLE
CDPS-897: add document service integration and get all photos for pri…

### PR DESCRIPTION
**Pull request by whitfield-mj**
…soner endpoint

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/controller/PrisonerPhotographController.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import uk.gov.justice.digital.hmpps.prisonperson.annotation.UsernameHeader
+import uk.gov.justice.digital.hmpps.prisonperson.client.documentservice.dto.DocumentDto
+import uk.gov.justice.digital.hmpps.prisonperson.client.documentservice.dto.DocumentRequestContext
+import uk.gov.justice.digital.hmpps.prisonperson.service.PhotographService
+import uk.gov.justice.hmpps.kotlin.common.ErrorResponse
+
+@RestController
+@ServiceNameHeader
+@ActiveCaseLoadIdHeader
+@UsernameHeader
```
Comment (whitfield-mj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/controller/PrisonerPhotographController.kt:29]: I've followed the pattern that the document service uses in terms of having headers for these things. It made sense to me to do it in the same way. Open to discussion

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/config/WebClientConfiguration.kt [hunk 2]
```diff
@@ -27,4 +28,8 @@ class WebClientConfiguration(
   @Bean
   fun prisonerSearchWebClient(authorizedClientManager: OAuth2AuthorizedClientManager, builder: WebClient.Builder) =
     builder.authorisedWebClient(authorizedClientManager, "prisoner-search-api", prisonerSearchBaseUri, timeout)
+
+  @Bean
+  fun documentServiceWebClient(authorizedClientManager: OAuth2AuthorizedClientManager, builder: WebClient.Builder) =
+    builder.authorisedWebClient(authorizedClientManager, "prisoner-search-api", documentServiceBaseUri, timeout)
```
Comment (brightonsbox) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/config/WebClientConfiguration.kt:34]: I seem to think that there was a trend for having separate client creds for each web client, and that's the way I've gone with our configuration so far.  So I think we need:
 * new set of client creds for document api creating in auth, with document service roles
 * the document service roles removing from the prisoner search client
 * this changing from `prisoner-search-api` to `document-service`
 * an entry in `application.yml` under  `spring.security.oauth2.client.registration`:
 ```
          document-service:
            provider: hmpps-auth
            client-id: ${document-service.client.id}
            client-secret: ${document-service.client.secret}
            authorization-grant-type: client_credentials
            scope: read
```

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/annotation/ActiveCaseLoadIdHeader.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package uk.gov.justice.digital.hmpps.prisonperson.annotation
```
Comment (brightonsbox) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/annotation/ActiveCaseLoadIdHeader.kt:1]: As discussed we've got a ticket to remove the header stuff as a follow up

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/client/documentservice/DocumentServiceClient.kt [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import uk.gov.justice.digital.hmpps.prisonperson.client.documentservice.dto.DocumentType
+import uk.gov.justice.digital.hmpps.prisonperson.client.documentservice.dto.OrderBy
+import uk.gov.justice.digital.hmpps.prisonperson.client.documentservice.dto.OrderByDirection
+import uk.gov.justice.digital.hmpps.prisonperson.config.DownstreamServiceException
+
+@Component
+class DocumentServiceClient(@Qualifier("documentServiceWebClient") private val webClient: WebClient) {
+  fun getPhotosForPrisoner(prisonerNumber: String, documentRequestContext: DocumentRequestContext): DocumentSearchResponseDto? {
+    val requestBody = DocumentSearchRequestDto(
+      documentType = DocumentType.PRISONER_PROFILE_PICTURE,
```
Comment (brightonsbox) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/client/documentservice/DocumentServiceClient.kt:24]: Really minor one but I typically do a further import

`import uk.gov.justice.digital.hmpps.prisonperson.client.documentservice.dto.DocumentType.PRISONER_PROFILE_PICTURE`

so it then reads: `documentType = PRISONER_PROFILE_PICTURE,`

makes it a bit more succinct and easier to read

**Review by brightonsbox — APPROVED** [review]
LGTM

**Review by brightonsbox — APPROVED** [review]
LGTM

**Pull request merged**